---
title: Analysis of RNA Sequencing data
authors:
- Malini Manoharan
tags:
- knowledge
- example
created_at: 2020-12-08
updated_at: 2020-12-08
tldr: This is short description of the content and findings of the post.
---

```{r setup, include=TRUE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
#Read the count count file
tcell_data <- read.csv("data/Exp6_Exp3_Counts_Test.csv", sep=",", header= TRUE,row.names = 1)
#Read the meta data file
tcell_meta <- read.csv("data/Metadata.csv", sep=",", header= TRUE)
#Deseq2 for data normalization
library(DESeq2)
library(pheatmap)
library(ggplot2)
library(cowplot)
```

The RNAseq Data provided is a time series data of T-cell activation across different donors in two different experiments.

## Data Normalization
All RNASeq analsis begins with a sample level QC using Principal Component Analysis (PCA) and hierarchical clustering methods. It helps us understand if the samples being analyed 
in the following aspects.

    Which samples are similar to each other, which are different?
    Does this fit to the expectation from the experiment’s design?
    What are the major sources of variation in the dataset?
    
sample-level QC allows us to see how well replicates cluster together, as well as, observe whether our experimental condition represents the major source of variation in the data. Performing sample-level QC can also identify any sample outliers, which may need to be explored further to determine whether they need to be removed prior to DE analysis.Normalized count data is used to compute the PCA. Normalization in this case is done using Deseq2. 

The SummarizedExperiment class is used to store rectangular matrices of experimental results, which are commonly produced by sequencing and microarray experiments. Each object stores observations of one or more samples, along with additional meta-data describing both the observations (features) and samples (phenotypes).


```{r Deseq2,message=FALSE}
dds <- DESeqDataSetFromMatrix(countData = tcell_data, colData = tcell_meta, design = ~ Time)
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=TRUE)
se_normalized <- SummarizedExperiment(assays=list(counts=normalized_counts), colData=tcell_meta)
dt_normalized <- DESeqTransform(se_normalized)
```
## Principal Component Analysis (PCA)

Principal Component Analysis (PCA) is a technique used to emphasize variation and bring out strong patterns in a dataset (dimensionality reduction).If two samples have similar levels of expression for the genes that contribute significantly to the variation represented by PC1, they will be plotted close together on the PC1 axis. Therefore, we would expect that replicates to have similar scores (since the same genes are changing) and cluster together on PC1 and/or PC2, and the samples from different time points/experiments to have different scores. 

```{r pcaraw, dpi = 200,fig.cap="Figure 1 : PCA of top normalized count data",message=FALSE}
plotPCA(dt_normalized, intgroup = "Batch", returnData = FALSE)
norm_dis <- dist(t(normalized_counts), method = "euclidean")
```

## Heirarchical Clustering

```{r hclustraw, dpi = 200,fig.cap="Figure 2: Heirarchical clustering of samples"}
norm_hc <- hclust(norm_dis, method = "ward.D")
plot(norm_hc,cex=0.5)
```
### Observations

The PCA plot and the heirarchical clustering suggests that samples from the different experiments are clusted together and therefore is indicative of batch effects coming from the different experimental setup.  

It was also observed that the pca does not change with the top 500/5000 genes suggesting a less noisy data. This can be reconfimed with the correlation heatmap (below) which suggests that the expression of the samples across the two experiments are highly correlated with respect to the different time points. 
```{r normtop500, fig.cap="Figure 3 : PCA of top 500(right) and top 5000 (left) genes",message=FALSE}
p1<-plotPCA(dt_normalized, intgroup = "Batch", ntop = 500, returnData = FALSE)
p2<-plotPCA(dt_normalized, intgroup = "Batch", ntop = 5000, returnData = FALSE)
plot_grid(p1, p2)
```

```{r heat, dpi = 300,message=FALSE, fig.cap="Figure 4 : Heatmap of samples based on  correlation of expression data"}
norm_mat <- assay(dt_normalized)
norm_cor <- cor(norm_mat)
fontsize_row = 10 - nrow(norm_cor) / 15
pheatmap(norm_cor,fontsize_row=fontsize_row,fontsize_col = fontsize_row)
```
## Batch Correction

The transformations implemented in DESeq2, vst and rlog, compute a variance stabilizing transformation which is roughly similar to putting the data on the log2 scale, while also dealing with the sampling variability of low counts. It uses the design formula to calculate the within-group variability.However it does not use the design to remove variation in the data. 

It is possible to visualize the transformed data with batch variation removed, using the removeBatchEffect function from limma. This simply removes any shifts in the log2-scale expression data that can be explained by batch. The paradigm for this operation for designs with balanced batches would be:

The above approach seemed to work better for the current dataset compared  to combat from the SVA package

```{r normalization using vst and limma,message=FALSE}
dds <- DESeqDataSetFromMatrix(countData=tcell_data, colData=tcell_meta, design = ~ Batch + Time)
dds <- DESeq(dds)
vsd <- vst(dds, blind=FALSE)
mat <- assay(vsd)
mat <- limma::removeBatchEffect(mat, vsd$Batch)
assay(vsd) <- mat
counts_batch_corrected <- assay(vsd)
se_normalized <- SummarizedExperiment(assays=list(counts=counts_batch_corrected), colData=tcell_meta)
dt_normalized <- DESeqTransform(se_normalized)
```
### Observations

The plots below indicate that the batch effects have been nullified.Clustering of the samples is now based on the experimental design. And the data can be compared across different time points capturing information across multiple donors. 

```{r pcabatch, dpi = 200, fig.cap="Figure 5 : PCA of batch effect removed data",message=FALSE}
plotPCA(dt_normalized, intgroup = "Time", returnData = FALSE)
```

```{r hclustbatch, dpi = 200, fig.cap="Figure 6:Heirarchical clustering after batch correction",message=FALSE}
combat_dis <- dist(t(counts_batch_corrected), method = "euclidean")
combat_hc <- hclust(combat_dis, method = "ward")
plot(combat_hc,cex=0.5)
```

## References

Love MI, Huber W, Anders S (2014). “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.” Genome Biology, 15, 550. doi: 10.1186/s13059-014-0550-8. 

Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), e47. doi: 10.1093/nar/gkv007. 
